reference/opt/shaders-msl/mesh/basic.msl30.spv14.mesh

```diff
@@ -0,0 +1,46 @@
+#pragma clang diagnostic ignored "-Wmissing-prototypes"
+
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using namespace metal;
+
+struct gl_MeshPerVertexEXT
+{
+    float4 gl_Position [[position]];
+    float gl_PointSize [[point_size]];
+};
+
+constant uint3 gl_WorkGroupSize [[maybe_unused]] = uint3(2u, 1u, 1u);
+
+struct spvPerVertex
+{
+    float4 gl_Position [[position]];
+    float gl_PointSize [[point_size]];
+};
+
+using spvMesh_t = mesh<spvPerVertex, void, 2, 2, topology::point>;
+
+static inline __attribute__((always_inline))
+void _4(threadgroup gl_MeshPerVertexEXT (&gl_MeshVerticesEXT)[2], thread uint& gl_LocalInvocationIndex, spvMesh_t spvMesh, thread uint3& gl_GlobalInvocationID)
+{
+    spvMesh.set_primitive_count(8u);
+    gl_MeshVerticesEXT[gl_LocalInvocationIndex].gl_Position = float4(1.0, 2.0, 3.0, 4.0);
+    gl_MeshVerticesEXT[gl_LocalInvocationIndex].gl_PointSize = 8.0;
+    spvMesh.set_index(gl_LocalInvocationIndex, gl_GlobalInvocationID.y);
+}
+
+[[mesh]] void main0(uint gl_LocalInvocationIndex [[thread_index_in_threadgroup]], uint3 gl_GlobalInvocationID [[thread_position_in_grid]], spvMesh_t spvMesh)
+{
+    threadgroup gl_MeshPerVertexEXT gl_MeshVerticesEXT[2];
+    _4(gl_MeshVerticesEXT, gl_LocalInvocationIndex, spvMesh, gl_GlobalInvocationID);
```

Comment:
What is the equivalent for SetMeshOutputsEXT in Metal?

Reply:
Similar to NV extension only `set_primitive_count`. Also index buffer is more similar to original NV extension(`uint index[max_prim*3]`).